### ISSUE TITLE
Link header doesn't support multiple links

### ISSUE THREAD

**Issue by alissapajer** [http4s/http4s#2852]
This is an example of the `Link` header returned from the github api. In this case, the `Link` header contents are used to support paginated api responses.

```
~\ε curl -I "https://api.github.com/search/code?q=addClass+user%3Amozilla&page=2"
HTTP/1.1 200 OK
Server: GitHub.com
Date: Mon, 09 Sep 2019 22:08:38 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 166756
Status: 200 OK
X-RateLimit-Limit: 10
X-RateLimit-Remaining: 9
X-RateLimit-Reset: 1568066978
Cache-Control: no-cache
X-GitHub-Media-Type: github.v3; format=json
Link: <https://api.github.com/search/code?q=addClass+user%3Amozilla&page=1>; rel="prev", <https://api.github.com/search/code?q=addClass+user%3Amozilla&page=3>; rel="next", <https://api.github.com/search/code?q=addClass+user%3Amozilla&page=34>; rel="last", <https://api.github.com/search/code?q=addClass+user%3Amozilla&page=1>; rel="first"
Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type
Access-Control-Allow-Origin: *
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
X-Frame-Options: deny
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
Content-Security-Policy: default-src 'none'
X-GitHub-Request-Id: F0D9:5F54:15E67:21DE4:5D76CD65
```

Observe that the `Link` header contains four urls, each with their own relation.

Consistent with this response, the [link header RFC](https://tools.ietf.org/html/rfc8288#section-3) specifies that one or more links can be serialized into the link header.

http4s only supports a single url in a parsed `Link` header. Furthermore, if you parse a header with multiple urls currently, it successfully parses into a `Link` using the first url in the string, silently discarding the subsequent urls.

In order for http4s to fully support (or at least more thoroughly support) real-world link headers, we need to support multiple urls. My first thought would be that we could wrap a `NonEmptyList[Link]` as a `Header`. And `Link` on its own would no longer be a viable `Header`.

This isn't a standalone example, but if you have a `val headers: Headers` in scope and you run
```scala
headers foreach {
  case h =>
    println(s"got header: $h")
}

headers foreach {
  case Link(l) =>
    println(s"got a link: $l")
  case _ => ()
} 
```
then you will see printed
```
got header: Link: <https://api.github.com/search/code?q=addClass+user%3Amozilla&page=1>; rel="prev", <https://api.github.com/search/code?q=addClass+user%3Amozilla&page=3>; rel="next", <https://api.github.com/search/code?q=addClass+user%3Amozilla&page=34>; rel="last", <https://api.github.com/search/code?q=addClass+user%3Amozilla&page=1>; rel="first"
// etc.
```
and
```
got a link: Link: <https://api.github.com/search/code?q=addClass%20user%3Amozilla&page=1>; rel=prev
```

**Comment by alissapajer** [comment]
I was able to work around this using `akka-http`:

```scala
import akka.http.scaladsl.model.HttpHeader
import akka.http.scaladsl.model.headers.{Link => AkkaLink, LinkValue}

HttpHeader.parse("link", header.value) match
  case HttpHeader.ParsingResult.Ok(AkkaLink(links), errors) =>
    links collectFirst {
      case LinkValue(uri, params) if ... => ...
    }
  }
```

**Comment by rossabaker** [comment]
Thank you for the detailed report, and sorry for not getting to it sooner.  There's an open PR, which I'll discuss over there, and we'll see if we can get this fixed for 0.21.
